gui/src/pages/NewProjectConfig.svelte

```diff
@@ -0,0 +1,150 @@
+<script lang="ts">
+  import Layout from '../components/Layout.svelte';
+  import Textbox from '../components/Textbox.svelte';
+  import Spinner from '../components/Spinner.svelte';
+  import ErrorLabel from '../components/ErrorLabel.svelte';
+  import SubmitButton from '../components/form/SubmitButton.svelte';
+  import CenterFormPanel from '../components/form/CenterFormPanel.svelte';
+  import { api, isClientError } from '../lib/api';
+  import * as router from 'svelte-spa-router';
+
+  export let params = { starter: '' };
+
+  const { starter } = params;
+
+  let name = starter;
+  let templateUrl: string | null = null;
+  let error: Error | null = null;
+
+  let workingDirectory: string | null = null;
+
+  const setWorkingDirectory = (dir: string) => {
+    workingDirectory = dir;
+  };
+
+  (async () => {
+    setWorkingDirectory(await api.kernel.getUserHomeDir());
+
+    if (starter !== 'empty') {
+      templateUrl = String(
+        (await api.kernel.describeTemplates()).find((x) => x.name === starter)
+          ?.url,
+      );
+    }
+  })();
+</script>
+
+<Layout>
+  <CenterFormPanel title={`New project: ${starter}`} backRoute="#/new-project">
+    <form
+      on:submit|preventDefault={async () => {
+        error = null;
+        let workspaceId;
+
+        try {
+          workspaceId = await api.kernel.createProject(
+            `${workingDirectory}/${name}`,
```

Comment:
The way I build the root directory in form submission here is kind of suspect. Seems to work but I know paths can be difficult - does it look alright to you @brandonbloom ?

Reply:
This doesn't do the right thing for the root directory. The API returned path will never have a trailing slash except for the root directory (where the trailing slash is the entire path).

Reply:
Should be fixed now